src/main/java/com/tananaev/adblib/AdbConnection.java

```diff
@@ -257,6 +276,19 @@ public int getMaxData() throws InterruptedException, IOException {
      * @throws InterruptedException If we are unable to wait for the connection to finish
      */
     public void connect() throws IOException, InterruptedException {
+        connect(false);
+    }
+
+    /**
+     * Same as {@link #connect()}, but also throws {@link AdbAuthenticationFailedException} if the
+     * peer rejects the first authentication attempt, which indicates that the peer has not saved
+     * our public key from a previous connection
+     */
+    public void connectIfAuthorised() throws IOException, InterruptedException, AdbAuthenticationFailedException {
```

Comment:
I think we can just expose `connect(boolean throwOnUnauthorised)` as public.

Reply:
Yeah fair enough. I've just added 20adca3 to make this change.